lib/internal/util/inspect.js

```diff
@@ -228,6 +231,33 @@ function getUserOptions(ctx) {
     getters: ctx.getters,
     ...ctx.userOptions
   };
+
+  // Typically, the target value will be an instance of `Object`. If that is
+  // *not* the case, the object may come from another vm.Context, and we want
+  // to avoid passing it objects from this Context in that case, so we remove
+  // the prototype from the returned object itself + the `stylize()` function,
+  // and remove all other non-primitives, including non-primitive user options.
+  if (isCrossContext) {
+    ObjectSetPrototypeOf(ret, null);
+    for (const key of ObjectKeys(ret)) {
+      if ((typeof ret[key] === 'object' || typeof ret[key] === 'function') &&
+          ret[key] !== null) {
+        delete ret[key];
+      }
+    }
+    ret.stylize = ObjectSetPrototypeOf((value, flavour) => {
+      let stylized;
+      try {
+        stylized = ctx.stylize(value, flavour);
+      } catch {}
```

Comment:
The current stylize function does not have such a guard. I would keep that aligned.

```suggestion
      const stylized = ctx.stylize(value, flavour);
```

Reply:
Not sure what you’re referring to here but passing random values exceptions leads to cross-context object passing.

Reply:
But this is about the error case? If the stylize function fails, it'll create a new error. If I'm not mistaken, that error would be created in the original VM context?

Reply:
Correct, that is the problem here.

Reply:
If I'm not mistaken, creating the error with the VM context is ideal, since it's not providing information from outside of the VM inside of the VM?

Reply:
```js
vm.runInNewContext(`
  ({
    [Symbol.for('nodejs.util.inspect.custom')](depth, ctx) {
      try {
        ctx.stylize(Symbol('A'));
      } catch (err) {
        // Currently aborts from within the context, not good
        err.constructor.constructor("return process")().abort();  
      }
    }
  })
`)
```